app/controllers/admin/tenants_controller.rb

```diff
@@ -1,5 +1,62 @@
 class Admin::TenantsController < Admin::BaseController
   def index
-    @tenants = Tenant.select(:organization, :location, :slug).all
+    @tenants = filtered_tenants
   end
+
+  def edit
+    @tenant = Tenant.find(tenant_id)
+  end
+
+  def update
+    tenant = Tenant.find(tenant_id)
+    old_approved_status = tenant.approved
+    tenant.update(params_update)
+
+    if old_approved_status == false && tenant.approved == true
+      tenant.users.each do |user|
+        UserMailer.approved_borrower(user, tenant).deliver_now
+      end
+
+    end
+    redirect_to admin_tenants_path
+  end
+
+  private
+
+  def tenant_id
+    params.require(:id)
+  end
+
+  def params_update
+    params.require(:tenant).permit(:organization, :location, :active, :approved)
+  end
+
+  def filtered_tenants
```

Comment:
We COULD convert this into a hash, but I think we've reached the point where we need to ship as opposed to making our code more clever.